### PR TITLE
feat: improve alibaba cloud credential providers in OSS artifacts

### PR DESCRIPTION
**Pull request by DahuK**
### Motivation

Improve the credential security of OSS artifacts, avoid hardcoding AK in secret
For more details abount alibaba cloud credential provider,  please refer to  [credentails-go](https://github.com/aliyun/credentials-go/blob/master/README.md)

### Modifications

add new configuration for OSSBucket in `workflow_types.go` to support using the default provider chain or the oidc token provider to get alibaba cloud credentials. 

```go
	// RoleARN is the Alibaba Cloud Resource Name(ARN) of the role to assume.
	RoleARN string `json:"roleARN,omitempty" protobuf:"bytes,8,opt,name=roleARN"`

	// OidcProviderARN is the Alibaba Cloud Resource Name (ARN) of the OIDC IdP.
	OidcProviderARN string `json:"OidcProviderARN,omitempty" protobuf:"bytes,9,opt,name=OidcProviderARN"`

	// OidcTokenFile is the file path of the OIDC token.
	OidcTokenFile string `json:"OidcTokenFile,omitempty" protobuf:"bytes,10,opt,name=OidcTokenFile"`

	// UseSDKCreds tells the driver to figure out credentials based on sdk defaults.
	UseSDKCreds bool `json:"useSDKCreds,omitempty" protobuf:"varint,11,opt,name=useSDKCreds"`
```
and refine the artifacts `oss.go` to init the OSS client with then given provider configuration

### Verification
 - Run make pre-commit -B
 - Run draft PR to check changes

### CODE REVIEW

File: workflow/artifacts/oss/oss.go [hunk 1]
```diff
@@ -39,11 +51,84 @@ var (
...
+			oss.SetCredentialsProvider(provider))
+	} else if ossDriver.RoleARN != "" && ossDriver.OidcProviderARN != "" && ossDriver.OidcTokenFilePath != "" {
+		// using oidc provider for RRSA auth
+		log.Infof("Using oidc provider for OSS driver")
+		cred := newOidcCredential(ossDriver.RoleARN, ossDriver.OidcProviderARN, ossDriver.OidcTokenFilePath)
+		provider := &ossCredentialsProvider{cred: cred}
+		return oss.New(ossDriver.Endpoint, "", "",
+			oss.SetCredentialsProvider(provider))
+	}
+	//using ak sec
```
Comment (terrytangyuan) [inline workflow/artifacts/oss/oss.go:131]: Be more descriptive or remove the comment

**Review by terrytangyuan — COMMENTED** [review]
Have you tested this change in an actual OSS bucket?

**Comment by terrytangyuan** [comment]
Please also resolve conflicts

**Comment by DahuK** [comment]
> Have you tested this change in an actual OSS bucket?

@terrytangyuan  Thanks for your review!  I have tested both of RRSA service account mode and the original AK/SK mode in ACK cluster,  it works fine when saving and loading OSS artifacts in output/input wf steps.
 
image

image

File: workflow/artifacts/oss/oss.go (hunk 1 above)
Comment (DahuK) [reply workflow/artifacts/oss/oss.go]: fixed

**Comment by DahuK** [comment]
> Please also resolve conflicts

Fixed, thanks！and added some instructions on how to cofigure and use Alibaba Cloud OSS RRSA in [configure-artifact-repository.md](https://github.com/argoproj/argo-workflows/pull/11453/files#diff-dd51c81cd60665f9118bad3a1a3e05ecc24484f9eda386f66469dfafdb294282)

**Review by terrytangyuan — APPROVED** [review]
Thank you!

**Pull request merged**